tests/countries/test_sierra_leone.py

```diff
@@ -0,0 +1,194 @@
+            "New Year's Day",
+            (f"{year}-01-01" for year in range(1961, 2050)),
+        )
+
+    def test_armed_forces_day(self):
+        name = "Armed Forces Day"
+        self.assertNoHolidayName(name, range(1961, 2002))
+        self.assertHolidayName(
+            name,
+            (f"{year}-02-18" for year in range(2002, 2050)),
+        )
+
+    def test_international_womens_day(self):
+        name = "International Women's Day"
+        self.assertNoHolidayName(name, range(1961, 2018))
+        self.assertHolidayName(
+            name,
+            (f"{year}-03-08" for year in range(2018, 2050)),
+        )
+
+    def test_good_friday(self):
+        name = "Good Friday"
+        self.assertHolidayName(
+            name,
+            "2022-04-15",
+            "2023-04-07",
+            "2024-03-29",
+            "2025-04-18",
+            "2026-04-03",
+        )
+        self.assertHolidayName(name, range(1961, 2050))
+
+    def test_easter_monday(self):
+        name = "Easter Monday"
+        self.assertHolidayName(
+            name,
+            "2022-04-18",
+            "2023-04-10",
+            "2024-04-01",
+            "2025-04-21",
+            "2026-04-06",
+        )
+        self.assertHolidayName(name, range(1961, 2050))
+
+    def test_independence_day(self):
+        name = "Independence Day"
+        self.assertHolidayName(
+            name,
+            (f"{year}-04-27" for year in range(1961, 2050)),
+        )
+        dt = (
+            "2008-04-28",  # Sunday -> Monday
+            "2013-04-29",  # Saturday -> Monday
+            "2019-04-29",  # Saturday -> Monday
+            "2024-04-29",  # Saturday -> Monday
+        )
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_labor_day(self):
+        name = "International Worker's Day"
+        self.assertHolidayName(
+            name,
+            (f"{year}-05-01" for year in range(1961, 2050)),
+        )
+
+    def test_christmas_day(self):
+        name = "Christmas Day"
+        self.assertHolidayName(
+            name,
+            (f"{year}-12-25" for year in range(1961, 2050)),
+        )
+
+    def test_boxing_day(self):
+        name = "Boxing Day"
+        self.assertHolidayName(
+            name,
+            (f"{year}-12-26" for year in range(1961, 2050)),
+        )
+
+    def test_islamic_holidays(self):
+        self.assertHolidayName(
+            "Eid al-Fitr",
+            "2023-04-21",
+            "2024-04-10",
+            "2025-03-30",
+            "2026-03-20",
+        )
+        self.assertHolidayName(
+            "Eid al-Adha",
+            "2023-06-28",
+            "2024-06-16",
+            "2025-06-06",
+            "2026-05-27",
+        )
+        self.assertHolidayName(
+            "Prophet's Birthday",
+            "2023-09-27",
+            "2024-09-15",
+            "2025-09-04",
+            "2026-08-25",
+        )
```

Comment:
It's better to create separate methods for these holidays. Also no need to check futute dates unless it's really necessary. And need to check for holiday name existence on the entire range of years.